src/common/file-utils.cpp

```diff
@@ -34,21 +34,136 @@
 #include <stdio.h>
 #include "uniwin.h"
 #include <errno.h>
+#else
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#endif        // !_MSC_VER
+#ifdef _WIN32 // for rnp_mkstemp
 #include <locale>
 #include <codecvt>
 #include <random>
+#endif
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
+#include <rnp/rnp_sdk.h>
+#ifdef _WIN32
+#include <locale>
+#include <codecvt>
+#endif
+
+extern "C" {
+int
+rnp_unlink(const char *filename)
+{
+#ifdef _WIN32
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8conv;
```

Comment:
Here and in the functions below:
Shouldn't we wrap C++ code in try..catch so no exception will go outside?

Reply:
Well, it's hard to define what `errno` these function should set in this case to mimick ordinary file functions...

Reply:
@rrrooommmaaa I don't think that we check `errno` in code, just success/failure. As this. code is used only by us we may freely skip setting errno or set it to some specific value, like ENOMEM

Reply:
Done